### ISSUE TITLE
useQuery `pollInterval` stop working in React v18 StrictMode

### ISSUE THREAD

**Issue by pobch** [apollographql/apollo-client#9819]
**Intended outcome:**
`pollInterval` should work either with React v18 StrictMode or without StrictMode

**Actual outcome:**
`pollInterval` stop working when using it with React v18 StrictMode

**How to reproduce the issue:**
1. Go to https://codesandbox.io/s/heuristic-bohr-df8gf9?file=/src/index.js
2. Open the dev tools, go to network tab
3. BUG: There is no new request to the graphql endpoint even though the `pollingInterval` is set to `2000`
4. If you remove `<React.StrictMode>` in `index.js` file, the polling will work as expected.

**Comment by cdnicoll** [comment]
+1 Seeing the same issue and confirmed that step 4 worked, but not an ideal solution

**Comment by boyerst** [comment]
Experiencing this as well. Manually triggering the poll via startPolling is a workaround

**Comment by gaswirth** [comment]
+1!

**Comment by protyze** [comment]
Experiencing the same.

**Comment by sanderkranz** [comment]
Same for me, `pollInterval` doesn't work. Fortunately, there are `startPolling` and `stopPolling` as a workaround:
```
const {startPolling, stopPolling} = useSessionCheckQuery()

useEffect(() => {
  startPolling(POLL_INTERVAL_MS)

  return () => {
    stopPolling()
  }
}, [])
```

**Comment by Maxou44** [comment]
Same here 🥲

**Comment by angelhodar** [comment]
Same, in my case its not working even if i call startPolling manually in useEffect. I have also set reactStrictMode to false in next.config.js and it doesnt solve the issue

**Comment by angelhodar** [comment]
@thiagokpelo Mmmm then i dont understand why it doesnt work in my case. I should try a minimal reproduction example. Thanks for telling me!!

**Comment by thiagokpelo** [comment]
@angelhodar will update us here if it resolves. If I can help just say! 👊

**Comment by angelhodar** [comment]
@thiagokpelo I dint know what i am doing wrong but with a fresh nextjs project and the last version of apollo client i cant get the poll interval to work. What versions did you use to debug your nextjs project?

**Comment by jpmermoz** [comment]
Same issue here. Next js with strict mode disabled and still not working

**Comment by thiagokpelo** [comment]
@angelhodar I'm using
```json

 "dependencies": {
    "@apollo/client": "^3.7.0",
    "@chakra-ui/react": "^2.3.5",
    "@datadog/browser-logs": "^4.21.2",
    "@datadog/browser-rum": "^4.21.2",
    "@emotion/react": "^11",
    "@emotion/styled": "^11",
    "@react-oauth/google": "^0.2.8",
    "apollo-link-rest": "^0.9.0-rc.1",
    "framer-motion": "^6",
    "graphql": "^16.6.0",
    "graphql-ws": "^5.11.2",
    "next": "12.3.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-intl": "^6.1.2"
  }
```

Have you tried to run the built project and with `npm start`?

**Comment by siegfried** [comment]
Same issue here. Maybe it is dedicated to local Nextjs server `yarn dev`, because my app on Vercel seems fine. Manually `startPolling` works.

**Comment by jurgelenas** [comment]
We are experiencing the same issue at https://github.com/ExpressLRS/ExpressLRS-Configurator

**Comment by MrDesjardins** [comment]
I had the same issue using `@apollo/client:3.7.1` and this new `3.7.2` version using CRA (React version 18.0.0).

Using a timer fixed the issue temporary.

Example of the fix:
```
const { refetch } = useMyEntity({
    onCompleted(data) {
       // Snip
    });
   // snip
   useEffect(() => {
    const refetchQueryInterval = setInterval(() => {
      refetch();
    }, 5000;
    return () => clearInterval(refetchQueryInterval);
});
```

**Comment by afonsomatos** [comment]
I stumbled in this issue as well.

**Comment by IShinkarenko** [comment]
The same issue with `Next.js 13.0.0`
I set `reactStrictMode` to `false` but that does not help

**Comment by ramblehead** [comment]
Similar issue with @apollo/client@^3.7.3 and next@13.1.1 (just updated)
```next start``` works as expected; ```next dev``` fails to start polling.

Using the following temp fix as others have suggested:
```ts
const {
  data: indexQuery,
  error: indexError,
  startPolling,
} = useQuery(IndexDocument, {
  ssr: false,
  // The following line is removed
  // pollInterval: 500,
});

// The following useEffect is added
useEffect(() => {
  startPolling(500);
}, [startPolling]);
```

**Comment by MarkAPhillips** [comment]
I tried version 3.7.4 but this seems to still be an issue - is they any idea as when this will be fixed ?

**Comment by mysticek** [comment]
`next@13.2.1` and `@apollo/client@3.7.9` , still not working, any news?

**Comment by patrick91** [comment]
@mysticek I think this is closely related to https://github.com/apollographql/apollo-client/issues/10540#issuecomment-1456300209 the workaround suggested there worked for polling too (which is similar to what @ramblehead did)

Hopefully, we find a good way to fix this soon 😊

**Comment by phryneas** [comment]
This is fixed in #10629 and will hopefully land in the next patch release.
